pkg/storage/docstore/docstore.go

```diff
@@ -0,0 +1,36 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package docstore
+
+import (
+	"github.com/hyperledger/aries-framework-go/pkg/doc/did"
+)
+
+// Connection is a pairwise relation between two DIDs (ours and theirs)
+type Connection struct {
+	Mine   *did.Doc
+	Theirs *did.Doc
+	Label  string
+}
+
+// Provider storage provider interface
+type Provider interface {
+	GetStoreHandle(name string) (Store, error)
+	Close()
+}
+
+// Store is the storage interface for DIDs
+type Store interface {
+	// Put stores the connection
+	Put(conn *Connection) error
+
+	// GetAll fetches all the stored connections
+	GetAll() (*[]Connection, error)
```

Comment:
This should really be an iterator. Enterprise agents may have hundreds of connections

Reply:
Doc store doesnt have getall function now. Added iterator in connection store based on updated design #22